crates/fluvio/src/config/cluster.rs

```diff
@@ -58,6 +62,74 @@ impl FluvioConfig {
         self.tls = tls.into();
         self
     }
+
+    pub fn query_metadata_path<'de, T>(&self, path: &str) -> Option<T>
+    where
+        T: Deserialize<'de>,
+    {
+        let mut metadata = self.metadata.as_ref().unwrap();
```

Comment:
This is not good. If there is no metadata, the querying here will shut my process.

Reply:
ohno, you're right, I missed that one!